lib/js/components/Layouts/ThreeColumnLayout/ThreeColumnLayout.vue

```diff
@@ -0,0 +1,239 @@
+<template>
+	<div class="layout">
+		<div
+			class="layout__overlay"
+			:class="{
+				'-visible': rightColumnVisibleLocal || leftColumnVisibleLocal,
+			}"
+		/>
+		<div
+			class="layout__leftColumn"
+			:class="{
+				'-desktopVisible': leftColumnVisible || leftColumnVisibleLocal,
+				'-mobileVisible': leftColumnVisibleLocal,
+			}"
+		>
+			<slot name="leftColumn" />
+		</div>
+		<div class="layout__contentColumn">
+			<div
+				class="layout__content"
+				:class="{
+					'-noPadding': contentWithoutPadding,
+				}"
+			>
+				<slot />
+			</div>
+		</div>
+		<div
+			class="layout__rightColumn"
+			:class="{
+				'-medium': rightColumnSize === THREE_COLUMN_LAYOUT_RIGHT_COLUMN_SIZE.MEDIUM,
+				'-large': rightColumnSize === THREE_COLUMN_LAYOUT_RIGHT_COLUMN_SIZE.LARGE,
+				'-desktopVisible': rightColumnVisible || rightColumnVisibleLocal,
+				'-mobileVisible': rightColumnVisibleLocal,
+			}"
+		>
+			<slot name="rightColumn" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/media-queries';
+
+$left-column-width: 23vw;
+$left-column-min-width: 200px;
+$left-column-max-width: 400px;
+
+$content-column-max-width: 900px;
+
+$right-column-medium-max-width: 400px;
+$right-column-medium-l-width: 30vw;
+$right-column-medium-l-min-width: 320px;
+$right-column-medium-l-max-width: 400px;
+$right-column-large-max-width: 560px;
+$right-column-large-l-width: 30vw;
+$right-column-large-l-min-width: 320px;
+$right-column-large-l-max-width: 560px;
+
+.layout {
+	$root: &;
+
+	display: flex;
+	flex-wrap: nowrap;
+	height: 100%;
+	overflow: hidden;
+	position: relative;
+	width: 100%;
+
+	&__overlay {
+		background: $color-default-overlay;
+		display: none;
+		height: 100%;
+		position: absolute;
+		width: 100%;
+
+		@media #{breakpoint-s()} {
+			&.-visible {
+				display: block;
+			}
+		}
+
+		@media #{breakpoint-l()} {
+			&.-visible {
+				display: none;
+			}
+		}
+	}
+
+	&__leftColumn {
+		display: none;
+		height: 100%;
+		left: 0;
+		position: absolute;
+		top: 0;
+		width: 100%;
+
+		@media #{breakpoint-l()} {
+			&.-desktopVisible {
+				display: initial;
+			}
+		}
+
+		@media #{breakpoint-s()} {
+			max-width: $left-column-max-width;
+			min-width: $left-column-min-width;
+			width: $left-column-width;
+		}
+
+		@media #{breakpoint-l()} {
+			height: auto;
+			left: initial;
+			position: initial;
+			top: initial;
+		}
+
+		&.-mobileVisible {
+			display: initial;
+		}
+	}
+
+	&__contentColumn {
```

Comment:
there is no way to scroll content in this container. I'm wondering if we need to fix it also for other columns or not?